packages/react/src/components/Storage/FileUploader/Previewer/index.tsx

```diff
@@ -29,7 +29,15 @@ export function Previewer({
   isSuccess,
   percentage,
   onFileClick,
+  isEditingNames,
+  saveEdit,
+  cancelEdit,
+  startEdit,
```

Comment:
Really concerned abt the amount of props being passed to this component. Either we are not handling enough logic in the parent, or we need multiple components with clear separations of concerns

Reply:
I'm trying to keep with the presentational design component strategy. Thus, we need a lot of logic to be passed into Previewer, since it has the Tracker component in it, that has lots of logic too. I"d be up for some suggestions on what would be the right approach to break this up. Let's look at it in a separate PR though.